src/main/java/com/denizenscript/ddiscordbot/DiscordCommand.java

```diff
@@ -169,7 +235,114 @@ public void execute(ScriptEntry scriptEntry) throws CommandExecutionException {
+                    client_d.getOrCreatePMChannel(client_d.getUserByID(user.asLong())).sendMessage(message.asString());
+                });
+                break;
+            case ROLE:
+                if (user == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no user given!");
+                    return;
+                }
+                if (guild == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no guild given!");
+                    return;
+                }
+                if (role == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no role given!");
+                    return;
+                }
+                if (message == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no action given!");
+                    return;
+                }
+                if (!dDiscordBot.instance.connections.containsKey(id.asString())) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: unknown ID!");
+                    return;
+                }
+                client_d = dDiscordBot.instance.connections.get(id.asString()).client;
+                IGuild guild_d = client_d.getGuildByID(guild.asLong());
+                switch (message.toString().toLowerCase()) {
+                    case "remove":
+                        client_d.getUserByID(user.asLong()).removeRole(guild_d.getRoleByID(role.asLong()));
+                        break;
+                    case "add":
+                        client_d.getUserByID(user.asLong()).addRole(guild_d.getRoleByID(role.asLong()));
+                        break;
+                    default:
+                        dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: Invalid action given - " + message.toString());
+                        return;
+                }
+                break;
+            case RENAME:
+                if (user == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to rename: no user given!");
+                    return;
+                }
+                if (guild == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to rename: no guild given!");
+                    return;
+                }
+                if (!dDiscordBot.instance.connections.containsKey(id.asString())) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to rename: unknown ID!");
+                    return;
+                }
+                if (message == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to rename: no name given!");
+                    return;
+                }
+                client_d = dDiscordBot.instance.connections.get(id.asString()).client;
+                IGuild guild_e = client_d.getGuildByID(guild.asLong());

```

Comment:
what is the `_d` and `_e` and all that here meant to be?

Also don't declare variables in one case of a switch then use them in another...